ArduPlane/GCS_Plane.cpp

```diff
@@ -8,6 +9,11 @@ uint8_t GCS_Plane::sysid_this_mav() const
 
 void GCS_Plane::update_vehicle_sensor_status_flags(void)
 {
+    // Compass-less plane force reporting AHRS healthy while on ground after arming
+    const auto &compass = AP::compass();
+    if (!plane.is_flying() && !compass.use_for_yaw(compass.get_first_usable())) {
```

Comment:
we'll need something more nuanced than this - this would stop errors for vehicles with GPS yaw for example

Reply:
@tridge what if I use the exact same test as the prearm that allowed the arm, only extend it while on  the ground?
ekf_type() != active_EKF_type() && !AP::compass().use_for_yaw() && !is_flying()

does moving baseline yaw failures switch the EKF type?